onnxruntime/core/optimizer/transpose_optimizer/optimizer_api_impl.cc

```diff
@@ -907,9 +908,9 @@ Status TransformLayoutForEP(Graph& graph, bool& modified, const IExecutionProvid
         onnx_layout_transformation::WrapTransposesAroundNode(*api_graph, *node, input_perms, {&output_perm});
       } else {
         onnx_layout_transformation::WrapTransposesAroundNode(*api_graph, *node, {&input_perm}, {&output_perm});
+        onnx_layout_transformation::SwapNodeOpTypeAndDomain(*api_graph, *node, node->OpType(), kMSInternalNHWCDomain);
```

Comment:
I would expect that ROCm/CUDA EPs treat `Resize` as layout insensitive, just like what CPU/XNNPACK/NNAPI did. These EPs recognize data format by `scaling/outputshape`. Otherwise, we have EPs can't run Resize with `internal domain` , and the others can only run within `internal domain`. It's quite divisive, isn't it?

Reply:
I don't think any change is required in this file. If we wrap the node in transposes it doesn't matter what the node was - it is now getting NHWC input and must to be in the internal nhwc domain. i.e. the call to SwapNodeOpTypeAndDomain is required as soon as you call WrapTransposesAroundNode.

FWIW the Resize special casing here will probably never be hit. Resize is only considered layout sensitive for CUDA/ROCm (which is temporary and meant to be being addressed). CUDA/ROCm do not request NHWC layout.